### PR TITLE
[3] Sets up redux, local storage and redux dev tools

### PR DESCRIPTION
**Pull request by comountainclimber**
https://github.com/comountainclimber/rn-task-manager/issues/3

- Adds and configures `redux` for state management. 
- Sets up `redux-devtools-extension` for use with `react-native-debugger` as described [here](https://github.com/jhen0409/react-native-debugger/blob/master/docs/redux-devtools-integration.md). You can view it in action below:
time-travel
- Sets up `redux-persist` configured with react-native's `AsyncStorage` module such that tasks persist to local storage across sessions.
storage-persistence

#### UNRELATED UX ENHANCEMENT:
This PR also fixed an issue that was bugging me related to the `<TaskInput/>` component. Now users will submit descriptions for tasks when hitting "return" previously the only option was to tap the plus button.
submit-on-return

#### NOTES:
- Basically all of the interfacing between redux and the application occurs in the newly created `screens/main/index.js` in attempt to keep concerns separate.
- `redux-persist` allows us to configure which reducers we want to persist (in the future we may not want to persist all global application state)
- Storage can be purged by invoking `persistor.purge()` (I was dropping this into `configureStore.js` during local testing

### CODE REVIEW

**Comment by comountainclimber** [comment]
@priyadarshy @bone-saw @brunobernardino mentioning you guys here because I cant seem to assign non-collaborators as reviewers

File: confgureStore.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+export const INITIAL_STATE = {
+  tasks: {}
+};
+
+const persistConfig = {
+  key: "root",
+  storage: AsyncStorage
+};
+
+const persistedReducer = persistReducer(persistConfig, rootReducer);
```
Comment (bone-saw) [inline confgureStore.js]: Cool, haven't worked with redux-persist before. Seems like this would give us 'out of the box' support for offline read-mode?

Not necessary for the scope of this demo, but a question to consider: if we have redux-persist as our persistent (including offline) storage of tasks, what other steps would we need to take in order to ensure actions taken while offline get properly synced when reconnecting?

File: reducers/taskReducer.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    case TOGGLE_COMPLETE_TASK: {
+      const newTasks = { ...state };
+      newTasks[action.selectedDate].find(
+        t => t.id === action.task.id
+      ).completed = action.task.completed;
+      return {
+        ...state,
+        ...newTasks
+      };
+    }
```
Comment (bone-saw) [inline reducers/taskReducer.js]: This approach should work fine for the list view, where we're showing the list of tasks per day.

However, in the full production app, this would need to be altered. If you're viewing the 'details view' (the view that appears when you select a task from the list), you can perform actions from there as well (e.g. move, complete). From that details view, you aren't guaranteed a selectedDate - the same task might appear on multiple days.

Again, what you've done here is totally fine for the scope of the project, just letting you know about extra complexity that would need to be addressed were this to be implemented for reals

File: App.js [hunk 3]
```diff
@@ -18,11 +24,20 @@ export default class App extends React.Component {
       "Nunito-Bold": require("./assets/fonts/Nunito-Bold.ttf"),
       "Nunito-Black": require("./assets/fonts/Nunito-Black.ttf")
     });
-    this.setState({ fontLoaded: true });
+    setTimeout(() => {
+      this.setState({ fontLoaded: true });
+    }, 400);
```
Comment (bone-saw) [inline App.js]: This looks hacky. As a user, I wouldn't wait to wait an extra half second anytime I launched the app just so fonts could load.

That being said, I can appreciate that you likely would rather about other things though for the purpose of this project, and this seems like a fine corner to cut so you can show us more important things.

File: confgureStore.js (hunk 1 above)
Comment (comountainclimber) [reply confgureStore.js]: I have actually a bunch of ideas around how this could be feasible and worked out. Excited to share in our meeting

File: reducers/taskReducer.js (hunk 2 above)
Comment (comountainclimber) [reply reducers/taskReducer.js]: Thats a great point... I think basically all of these reducers could be refactored to look up task by id alone instead of looking under the `selectedDate` key which would support the functionality you are mentioning. 

Something to consider as a future enhancement

File: App.js (hunk 3 above)
Comment (comountainclimber) [reply App.js]: Definitely hacky!

Without this the fonts load in sub 100ms which causes the animation to appear as "just a flicker". 400ms allows for one full run of the `.gif` loading animation. I could get rid of this timeout to improve load times but I actually think the flicker makes for a worse user experience.

The ideal scenario would just be to display our splash screen until initial application render is 100% ready. I will look into this as a future enhancement

**Comment by comountainclimber** [comment]
Thanks for taking a look and for the review @bone-saw !

File: App.js (hunk 3 above)
Comment (priyadarshy) [reply App.js]: That makes a lot of sense. It can be a bad experience if a splash screen just flickers. I didn't realize that was the reason for this time out but it makes sense now!
